### PR TITLE
'/training' support Kmeans Algorithm

### PR DESCRIPTION
**Pull request by JackyXu-Cool**
Call KMeans algorithm on provided dataset when the user selects "Kmeans"

### CODE REVIEW

File: mlr_backend/run.py [hunk 1]
```diff
@@ -1,27 +1,36 @@
...
 def trainData():
     request_data = request.get_json()
-    print("request data is")
-    print(request_data)
-    # TODO: Use ML Models to train data and send back useful information
+    models = request_data.get('models')
+    params = request_data.get('params')
+    dataset = cache.get("dataset")
+    dataset_in_np_array = np.genfromtxt(StringIO(dataset),
+                                        delimiter=',', usecols=np.arange(0, 5))
```
Comment (ruokun-niu) [inline mlr_backend/run.py]: Why is usecols set to np.arange(0, 5)? Our datasets could have more than 5 columns

File: mlr_backend/run.py [hunk 2]
```diff
@@ -1,27 +1,36 @@
...
+    if ("KMeans" in models):
+        centroid, clusterAssessment = kMeans(
+            dataset_in_np_array, int(params['Number_of_Clusters']))
```
Comment (ruokun-niu) [inline mlr_backend/run.py:42]: Quick question: Should we include the accuracy calculation right now?

Comment (ruokun-niu) [reply mlr_backend/run.py:42]: Or we can just do a separate PR for that

**Comment by ruokun-niu** [comment]
LGTM! Just a few comments to address

File: mlr_backend/run.py (hunk 1 above)
Comment (JackyXu-Cool) [reply mlr_backend/run.py]: Resolved

File: mlr_backend/run.py (hunk 2 above)
Comment (JackyXu-Cool) [reply mlr_backend/run.py:42]: Thanks Tommy for the comment! I am not quite familiar with ML :( Wondering what is the accuracy calculation and how we can do that.

Comment (ruokun-niu) [reply mlr_backend/run.py:42]: As discussed offline, we are moving this calculation to a separate PR as there is quite a bit of work

**Review by ruokun-niu — APPROVED** [review]

**Pull request merged**